# Would you include all suggested variants in the experiment (Connect, Learn, Help, Services)?
No. Since the representatives of all interviewed user groups ranked Help -> Services  -> rest, I would focus on these two alternatives in order to reduce the number of categories for the upcoming experiment to compare with the SQ. 

sophomore:
Help, Services, Interact, Connect, Learn
junior:
Help is the best, then Services too
senior:
 Help, Services, Connect, Interact, Learn

# What is the “business value” that performing this experiment would add within the broader strategy of the University?
Although (as the heatmap shows) users mostly use the university mainpage and the catalogue buttons (as to be expected on a uni-lib), from an SQ point of view it is hard to assess whether users need more help and don't find it since it is hidden under "interact" or Montana simply has competent users. Anyways, the University of Montana could get an opportunity of improving the library's User Experience hence improving the overall satisfaction of users and in the long run better ratings in the future and eventually more students.

# Which main metric would you choose to measure the success of a variant and perform the experiment on?
Main metric would be the CTR of the changed buttons and an assessment if that changed led to a more frequent use.

# Which additional metrics would you choose to track?
The basic addition would be the overall user-number on the library's website. In order to assess whether people were using the changes because they were looking for help/additional informations i would track the follow-up-CTR (links to Reference Services, Instruction Services, Subject Liaisons, Writing Center, About, Staff Directory, Library FAQ, Give to the Library, and Floor Maps)

# How would you define the null and the alternative hypotheses?
H0: the CTR of changed button equals the SQ
H1: the CTR of changed button changes

# What threshold for statistical significance would you set?
Since the general user seems to navigate the library's website with ease and the UX-Upgrade is easily implemented: alpha = 0.1 would be enough

# What is the minimum detectable effect (the smallest improvement you would care about) that you expect to detect?
All improvements that fall short of reaching the modest significance level set before would be "ignored".

# Do you think this experiment would require a software engineering team to develop a custom platform, or could it be developed with external tools such as Google Optimize?
Since it has only to be made sure, that the experiment is strictly randomized and all we care about is a changed button and no further changes will be applied, the task at hand should be easily implemented by the IT-team using google optimize.

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

filepath_1 = r"C:\Users\danie\Desktop\WBS.DataScience_Bootcamp\Week_5_AB_Testing\University of Montana\Connect.csv"
filepath_2 = r"C:\Users\danie\Desktop\WBS.DataScience_Bootcamp\Week_5_AB_Testing\University of Montana\Services.csv"
filepath_3 = r"C:\Users\danie\Desktop\WBS.DataScience_Bootcamp\Week_5_AB_Testing\University of Montana\Learn.csv"
filepath_4 = r"C:\Users\danie\Desktop\WBS.DataScience_Bootcamp\Week_5_AB_Testing\University of Montana\Interact.csv"
filepath_5 = r"C:\Users\danie\Desktop\WBS.DataScience_Bootcamp\Week_5_AB_Testing\University of Montana\Help.csv"

connect = pd.read_csv(filepath_1, sep = ",")
services = pd.read_csv(filepath_2, sep = ",")
learn = pd.read_csv(filepath_3, sep = ",")
interact = pd.read_csv(filepath_4, sep = ",")
help = pd.read_csv(filepath_5, sep = ",")

pd.set_option('display.float_format', lambda x: '%.2f' % x)
pd.set_option('display.max_rows', 1000)
pd.set_option('max_colwidth', 400)

## actual metrics

    Click-through rate (CTR) for the homepage. Amount of clicks on the button divided by the total visits to the page. Selected as a measure of the initial ability of the category title to attract users.
    Drop-off rate for the category pages. Percentage of visitors who leave the site from a given page, selected as a measure of the ability of the category page to meet user expectations.
    Homepage-return rate for the category pages. Percentage of users who navigated from the library homepage to the category page, then returned back to the homepage. This sequence of actions provides clues as to whether a user discovered the desired option on the category page; if not, the user would likely return to the homepage to continue navigation. Homepage-return rate was therefore selected as a measure of the ability of the category page to meet user expectations.


In [2]:
c = connect.copy()
c.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 58 entries, 0 to 57
Data columns (total 6 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Element ID            58 non-null     int64 
 1   Tag name              58 non-null     object
 2   Name                  58 non-null     object
 3   No. clicks            58 non-null     int64 
 4   Visible?              58 non-null     bool  
 5   Snapshot information  2 non-null      object
dtypes: bool(1), int64(2), object(3)
memory usage: 2.4+ KB


In [3]:
c["Snapshot information"]

# 2742 visits, 1587 clicks --> dropoff = 
2742-1587

1155

In [4]:
c_metrics = c[(c.Name == "CONNECT") | (c.Name == "Montana State University Libraries - Home")].reset_index()
c_metrics.drop("Element ID", axis=1, inplace=True)
c_metrics.drop("Tag name", axis=1, inplace=True)
c_metrics.drop("Visible?", axis=1, inplace=True)
c_metrics.drop("Snapshot information", axis=1, inplace=True)

In [5]:
c_metrics.drop("index", axis=1, inplace=True)

In [6]:
c_metrics.iloc[0,0] = "Home"

In [7]:
c_metrics.loc[2] = ["dropoff", 1155]
c_metrics.loc[3] = ["total", 2742]
c_metrics

Name  No. clicks
0     Home          83
1  CONNECT          53
2  dropoff        1155
3    total        2742

In [8]:
s = services.copy()

In [9]:
s["Snapshot information"]
# 2064 visits, 1348 clicks
2064 - 1348

716

In [10]:
s_metrics = s[(s.Name == "SERVICES") | (s.Name == "Montana State University - Home")].reset_index()
s_metrics.drop("Element ID", axis=1, inplace=True)
s_metrics.drop("Tag name", axis=1, inplace=True)
s_metrics.drop("Visible?", axis=1, inplace=True)
s_metrics.drop("Snapshot information", axis=1, inplace=True)

In [11]:
s_metrics.drop("index", axis=1, inplace=True)

In [12]:
s_metrics.iloc[0,0] = "Home"
s_metrics.loc[2] = ["dropoff", 716]
s_metrics.loc[3] = ["total", 2064]
s_metrics

Name  No. clicks
0      Home          49
1  SERVICES          45
2   dropoff         716
3     total        2064

In [13]:
l = learn.copy()

In [14]:
l["Snapshot information"]
#2747 visits 1652 clicks
2747-1652
# dropoff = 1095

1095

In [15]:
l_metrics = l[(l.Name == "LEARN") | (l.Name == "Montana State University - Home")].reset_index()
l_metrics.drop("Element ID", axis=1, inplace=True)
l_metrics.drop("Tag name", axis=1, inplace=True)
l_metrics.drop("Visible?", axis=1, inplace=True)
l_metrics.drop("Snapshot information", axis=1, inplace=True)

In [16]:
l_metrics.drop("index", axis=1, inplace=True)


In [17]:
l_metrics.iloc[0,0] = "Home"
l_metrics.loc[2] = ["dropoff", 1095]
l_metrics.loc[3] = ["total", 2747]
l_metrics

Name  No. clicks
0     Home          83
1    LEARN          21
2  dropoff        1095
3    total        2747

In [18]:
i = interact.copy()
i["Snapshot information"]
#10283 visits, 3714 clicks
10283 - 3714
#6569 dropoff


6569

In [19]:
i_metrics = i[(i.Name == "INTERACT") | (i.Name == "Montana State University - Home")].reset_index()
i_metrics.drop("Element ID", axis=1, inplace=True)
i_metrics.drop("Tag name", axis=1, inplace=True)
i_metrics.drop("Visible?", axis=1, inplace=True)
i_metrics.drop("Snapshot information", axis=1, inplace=True)

In [20]:
i_metrics.drop("index", axis=1, inplace=True)


In [21]:
i_metrics.iloc[0,0] = "Home"
i_metrics.loc[2] = ["dropoff", 6569]
i_metrics.loc[3] = ["total", 10283]
i_metrics

Name  No. clicks
0      Home        1291
1  INTERACT          42
2   dropoff        6569
3     total       10283

In [22]:
h = help.copy()
#h["Snapshot information"]
#3180 visits, 1717 clicks
3180-1717
#dropoff: 1463

1463

In [23]:
h["Snapshot information"]

0              Homepage Version 4 - Help   •   http://www.lib.montana.edu/index4.php
1     created 5-29-2013   •   20 days 4 hours 59 mins   •   3180 visits, 1717 clicks
2                                                                                NaN
3                                                                                NaN
4                                                                                NaN
5                                                                                NaN
6                                                                                NaN
7                                                                                NaN
8                                                                                NaN
9                                                                                NaN
10                                                                               NaN
11                                                               

In [24]:
h.groupby("Name").agg({"No. clicks": "sum"}).sort_values("No. clicks", ascending=False)

No. clicks
Name                                                                                                            
FIND                                                                                                         631
s.q                                                                                                          364
lib.montana.edu/find/                                                                                        139
Montana State University - Home                                                                              122
REQUEST                                                                                                       72
Hours                                                                                                         68
Search                                                                                                        64
HELP                                                                                                          38
News                                                                                                          27
Advanced Search                                                                                               20
Contact Us                                                                                                    17
Ask A Librarian                                                                                               15
lib.montana.edu/request/                                                                                      15
slideshow-right                                                                                               14
nav-item                                                                                                      11
Twitter                                                                                                       11
nav-item-dot                                                                                                  11
Tip of the Week: Submit a Request for the Library to Pu                                                        8
slideshow-left                                                                                                 8
lib.montana.edu/interact/                                                                                      7
6/17 Library Workshop: EndNote Web Basics                                                                      6
About MSU Library                                                                                              6
Find research materials, including articles, books, databases, journals, and course reserves                   4
Search Advanced Search FIND Find research materials, including articles, books, databases, journals,           4
Explore the world, explore the past Science & Natural History Filmmaking Archive MSU School of Film            4
Site Index & Site Search                                                                                       3
logo-ikd                                                                                                       3
.                                                                                                              3
logo                                                                                                           2
Montana State University Historical Photographs                                                                2
Indian Peoples of the Northern Great Plains                                                                    2
6/17 Library Workshop: Zotero Basics                                                                           2
s3.amazonaws.com/z90/d7563d332436cc96671be5ed9979147fx.png                                                     1
triangle-l                                                                                                     1
montana.summon.serialssolutions.co

In [25]:
h_metrics = h[(h.Name == "HELP") | (h.Name == "Montana State University - Home")].reset_index()
h_metrics.drop("Element ID", axis=1, inplace=True)
h_metrics.drop("Tag name", axis=1, inplace=True)
h_metrics.drop("Visible?", axis=1, inplace=True)
h_metrics.drop("Snapshot information", axis=1, inplace=True)

In [26]:
h_metrics.drop("index", axis=1, inplace=True)

In [27]:
h_metrics.iloc[0,0] = "Home"
h_metrics.loc[2] = ["dropoff", 1463]
h_metrics.loc[3] = ["total", 3180]
h_metrics

Name  No. clicks
0     Home         122
1     HELP          38
2  dropoff        1463
3    total        3180

In [28]:
montana_lib = pd.DataFrame()
montana_lib["interact"] = i_metrics["No. clicks"]
montana_lib["help"] = h_metrics["No. clicks"]
montana_lib["learn"] = l_metrics["No. clicks"]
montana_lib["services"] = s_metrics["No. clicks"]
montana_lib["connect"] = c_metrics["No. clicks"]
montana_lib["type"] = ["home", "click", "drop_off", "total"]
montana_test = montana_lib.copy()
montana_test = montana_test.set_index("type", drop=True).rename_axis(None)
montana_test.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4 entries, home to total
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   interact  4 non-null      int64
 1   help      4 non-null      int64
 2   learn     4 non-null      int64
 3   services  4 non-null      int64
 4   connect   4 non-null      int64
dtypes: int64(5)
memory usage: 192.0+ bytes


# Dataframe above accounts for the 3 predefined metrics in absolute numbers, thus creating rows with pct of total visits

In [29]:
montana = montana_test
montana.loc["CTR"] = ((montana.loc["click"]/montana.loc["total"])*100).round(2)
montana.loc["DO_rate"] = ((montana.loc["drop_off"]/montana.loc["total"])*100).round(2)
montana.loc["HP_rate"] = ((montana.loc["home"]/montana.loc["total"])*100).round(2)
montana

interact    help   learn  services  connect
home       1291.00  122.00   83.00     49.00    83.00
click        42.00   38.00   21.00     45.00    53.00
drop_off   6569.00 1463.00 1095.00    716.00  1155.00
total     10283.00 3180.00 2747.00   2064.00  2742.00
CTR           0.41    1.19    0.76      2.18     1.93
DO_rate      63.88   46.01   39.86     34.69    42.12
HP_rate      12.55    3.84    3.02      2.37     3.03

# What was the click-through rate for each version?
See DF above

# Which version was the winner?
"services" shows highest CTR while having lowest Homepage- and drop-off-ratings

# Do the results seem conclusive?
Although looking good, the findings need to be tested for statistical significance. Therefore a CHI-SQUARED-Test is needed since categorical data is under scrutiny

# Data-Requirements for CHI-SQUARED-Testing
- categorical data? both axes of DF show categorical values
- independence of observations? we need to trust crazy.eggs for setting up a valid research
- mutual exxclusiveness? given, since all categories are either/or
- sample size? needs to be checked due to dependence on EXPECTED VALUES

In [30]:
montana["total"] = montana.interact + montana.help + montana.learn + montana.services + montana.connect

#dropping percentages from DF since absolute numbers are to be examined in CHI-SQUARE-TEST / summing up percentages does not help here
montana_chi = montana.iloc[0:4, :]
montana_chi

interact    help   learn  services  connect    total
home       1291.00  122.00   83.00     49.00    83.00  1628.00
click        42.00   38.00   21.00     45.00    53.00   199.00
drop_off   6569.00 1463.00 1095.00    716.00  1155.00 10998.00
total     10283.00 3180.00 2747.00   2064.00  2742.00 21016.00

In [31]:
# For reasons of clarity, we get the largest index of the observed_expanded dataframe and assign it to a variable.
max_row_index = len(montana_chi.index)-1
max_row_index

3

In [32]:
# For reasons of clarity, we get the largest index of the observed_expanded dataframe columns and assign it to a variable.
max_column_index = len(montana_chi.columns)-1
max_column_index

5

In [33]:
# Create table for the expected values as a copy of the observed table.
# We will overwrite the values in the cells with the code below.
montana_chi2 = montana_chi.copy()

# Iterating over the rows in the table.
for i in range(montana_chi2.shape[0]):
  # Iterating over the columns in the table.
  for j in range(montana_chi2.shape[1]):
    # Setting the value in each cell to be equal to:
    # the Total value of that same columns in the observed_expanded table
    # (i.e. the total visitors of that version),
    # multiplied by the share of that row's total from the overall total
    # (i.e. the share of clicks/no-clicks from the overall total number of visitors) 
    montana_chi2.iloc[i,j] = montana_chi.iloc[max_row_index,j] * (montana_chi.iloc[i,max_column_index]/montana_chi.iloc[max_row_index,max_column_index])

montana_chi2

interact    help   learn  services  connect    total
home        796.57  246.34  212.80    159.89   212.41  1628.00
click        97.37   30.11   26.01     19.54    25.96   199.00
drop_off   5381.25 1664.14 1437.55   1080.12  1434.93 10998.00
total     10283.00 3180.00 2747.00   2064.00  2742.00 21016.00

# all expected values are larger than 5 --> sample size is sufficient

# DoF? 2*4 = 8 both totals are excluded from DoF calculation and formula is (rows-1)*(cloumns-1)
# Pearson-Residuals to be calculated now

In [34]:
# Define the function and show the dataframe with all the residuals.
def pearson_residuals(contingency_observed, contingency_expected):
    R = (contingency_observed - contingency_expected) / (np.sqrt(contingency_expected))
    return R

p_residuals = pearson_residuals(montana_chi, montana_chi2)
p_residuals

interact  help  learn  services  connect  total
home         17.52 -7.92  -8.90     -8.77    -8.88   0.00
click        -5.61  1.44  -0.98      5.76     5.31   0.00
drop_off     16.19 -4.93  -9.03    -11.08    -7.39   0.00
total         0.00  0.00   0.00      0.00     0.00   0.00

In [35]:
# Square all the residuals.
p_residuals_squared = p_residuals ** 2
p_residuals_squared

interact  help  learn  services  connect  total
home        306.89 62.76  79.17     76.90    78.84   0.00
click        31.49  2.07   0.97     33.16    28.15   0.00
drop_off    262.16 24.31  81.62    122.75    54.61   0.00
total         0.00  0.00   0.00      0.00     0.00   0.00

In [36]:
# Add all the squared residuals to get the test statistic.
test_statistic = p_residuals_squared.values.sum()
test_statistic

1245.8479430432371

In [37]:
residuals = []
columns = montana_chi.columns
for column in columns:
  for category in range(montana_chi.shape[0]):
    residuals.append(((montana_chi[column][category] - montana_chi2[column][category]) / np.sqrt(montana_chi2[column][category]))**2)

sum(residuals)

1245.847943043237

# critical value for CHI-SQUARED for alpha = 0.1 and DoF = 8: 13.36; test-statistic is by factor 100 bigger, so we are clearly in the critical zone and H0: all buttons are equal has to be rejected

# dropping the metrics for drop-off and home (since both measure the drop-off from original and not landing-page), although HUGE discrepencies between the diffenrent layouts can be observed

In [38]:
m = montana.copy()
m.drop("home", axis=0, inplace=True)
m.drop("drop_off", axis=0, inplace=True)
m.drop("CTR", axis=0, inplace=True)
m.drop("DO_rate", axis=0, inplace=True)
m.drop("HP_rate", axis=0, inplace=True)
m.loc["no_click"] = m.loc["total"] - m.loc["click"]
m.loc["visits"] = m.loc["total"]
m.drop("total", axis=0, inplace=True)

# showing click vs. non-click per button-design
m

interact    help   learn  services  connect    total
click        42.00   38.00   21.00     45.00    53.00   199.00
no_click  10241.00 3142.00 2726.00   2019.00  2689.00 20817.00
visits    10283.00 3180.00 2747.00   2064.00  2742.00 21016.00

# calculating expected values for m

In [39]:
m_test = m.iloc[0:-1, 0:-1]

In [40]:
m_test

interact    help   learn  services  connect
click        42.00   38.00   21.00     45.00    53.00
no_click  10241.00 3142.00 2726.00   2019.00  2689.00

In [41]:
m

interact    help   learn  services  connect    total
click        42.00   38.00   21.00     45.00    53.00   199.00
no_click  10241.00 3142.00 2726.00   2019.00  2689.00 20817.00
visits    10283.00 3180.00 2747.00   2064.00  2742.00 21016.00

In [42]:
# For reasons of clarity, we get the largest index of the observed_expanded dataframe and assign it to a variable.
max_row_index = len(m.index)-1
max_row_index

2

In [43]:
# For reasons of clarity, we get the largest index of the observed_expanded dataframe columns and assign it to a variable.
max_column_index = len(m.columns)-1
max_column_index

5

In [44]:
# Create table for the expected values as a copy of the observed table.
# We will overwrite the values in the cells with the code below.
m_chi = m.copy()

# Iterating over the rows in the table.
for i in range(m_chi.shape[0]):
  # Iterating over the columns in the table.
  for j in range(m_chi.shape[1]):
    # Setting the value in each cell to be equal to:
    # the Total value of that same columns in the observed_expanded table
    # (i.e. the total visitors of that version),
    # multiplied by the share of that row's total from the overall total
    # (i.e. the share of clicks/no-clicks from the overall total number of visitors) 
    m_chi.iloc[i,j] = m.iloc[max_row_index,j] * (m.iloc[i,max_column_index]/m.iloc[max_row_index,max_column_index])

m_chi

interact    help   learn  services  connect    total
click        97.37   30.11   26.01     19.54    25.96   199.00
no_click  10185.63 3149.89 2720.99   2044.46  2716.04 20817.00
visits    10283.00 3180.00 2747.00   2064.00  2742.00 21016.00

In [45]:
# Define the function and show the dataframe with all the residuals.
def pearson_residuals(contingency_observed, contingency_expected):
    R = (contingency_observed - contingency_expected) / (np.sqrt(contingency_expected))
    return R

p_residuals = pearson_residuals(m, m_chi)
p_residuals

interact  help  learn  services  connect  total
click        -5.61  1.44  -0.98      5.76     5.31   0.00
no_click      0.55 -0.14   0.10     -0.56    -0.52   0.00
visits        0.00  0.00   0.00      0.00     0.00   0.00

In [46]:
# Square all the residuals.
p_residuals_squared = p_residuals ** 2
p_residuals_squared

interact  help  learn  services  connect  total
click        31.49  2.07   0.97     33.16    28.15   0.00
no_click      0.30  0.02   0.01      0.32     0.27   0.00
visits        0.00  0.00   0.00      0.00     0.00   0.00

In [47]:
test_statistic = p_residuals_squared.values.sum()
test_statistic

96.74323537983281

In [48]:
residuals = []
columns = montana_chi.columns
for column in columns:
  for category in range(m.shape[0]):
    residuals.append(((m[column][category] - m_chi[column][category]) / np.sqrt(m_chi[column][category]))**2)

sum(residuals)

96.74323537983281

## again: chi-squared is huge, so it is obvious that H0 - all buttons perform equally well - has to be rejected.

# Since "interact" is clearly the worst performing button in terms of CTR it will be dropped from further analysis. Dropping the worst performer will be continued after repeated analysis of the remaining options

In [49]:
m_4 = m.copy()
m_4.drop("interact", axis=1, inplace=True)


In [50]:
m_4

help   learn  services  connect    total
click      38.00   21.00     45.00    53.00   199.00
no_click 3142.00 2726.00   2019.00  2689.00 20817.00
visits   3180.00 2747.00   2064.00  2742.00 21016.00

In [57]:
m_4_chi = m_4.iloc[0:-1, 0:-1]
m_4_chi

help   learn  services  connect
click      38.00   21.00     45.00    53.00
no_click 3142.00 2726.00   2019.00  2689.00

# instead of calculating chi-squared and expected values manually - the next steps will utilize scypy in order to perform the statistical analysis


In [51]:
from scipy.stats import chi2_contingency

In [58]:
chisq, pvalue, df, expected = chi2_contingency(m_4_chi)
chisq

22.45097953040183

In [59]:
expected

array([[  46.51635144,   40.1825212 ,   30.19174509,   40.10938228],
       [3133.48364856, 2706.8174788 , 2033.80825491, 2701.89061772]])

In [55]:
df

8

In [60]:
pvalue

5.255098702285646e-05

## p-value extremely small while chi-square clearly above critical value (for dof=3 (rows-1)*(columns-1) and alpha=0.1: crit_value = 6.251) - H0 has to be rejected
"learn" will be excluded next, because of having the worst CTR

In [63]:
m_3 = m_4_chi.copy()
m_3.drop("learn", axis=1, inplace=True)


In [64]:
m_3

help  services  connect
click      38.00     45.00    53.00
no_click 3142.00   2019.00  2689.00

In [65]:
chisq, pvalue, df, expected = chi2_contingency(m_3)

In [67]:
pd.DataFrame(expected)

0       1       2
0   54.15   35.15   46.70
1 3125.85 2028.85 2695.30

In [68]:
pvalue

0.013726659948517513

In [69]:
chisq

8.57683071094785

In [70]:
df

2

## pvalue smaller than alpha = 0.1 and chi-squared bigger than crit value (0.1, dof=2: 4.605), H0 to be rejected, difference in outcome did not happen by chance under the assumption of all buttons working equally fine
## since "help" is underperforming (CTR < expected value) it is discarded and we will finally compare "connect" and "services"

In [71]:
m_2 = m_3.copy()
m_2.drop("help", axis =1, inplace=True)

In [72]:
m_2

services  connect
click        45.00    53.00
no_click   2019.00  2689.00

In [74]:
chisq, pvalue, df, expected = chi2_contingency(m_2)

In [75]:
pd.DataFrame(expected)

0       1
0   42.09   55.91
1 2021.91 2686.09

## when comparing the two options "services" gets more clicks than expected while "connect" falls short of the expected value - but is that finding statisticlly significant?
- Chi squared test is run under the H0-assumption "services = connect"

In [76]:
df

1

In [77]:
chisq

0.24744894078894586

In [78]:
pvalue

0.6188771123975272

# chisq is NOT bigger than the crit_value (0.1, dof=1: 2.706) while the pvalue is bigger than alpha. H0 cannot be rejected in this case - so the difference between services and connect can result from chance. Since both perform way better than the other options the additional data given outside the dataset will be used to assess the performance of the two. The "no_click" is not important at the following stage since the user-behavior after conversion is relevant. Important to remember that the additional metrics take into account the drop-off-rates and the homepage-return-rates in order to assess the question whether the user's expectations were met.

In [80]:
m_f = m_2.copy()
m_f.drop("no_click", axis=0, inplace=True)

In [81]:
m_f

services  connect
click     45.00    53.00

In [84]:
m_f.loc["drop_off"] = [0, 6]
m_f.loc["hp_return"] = [0, 8]

In [85]:
m_f

services  connect
click         45.00    53.00
drop_off       0.00     6.00
hp_return      0.00     8.00

In [86]:
14/53*100

26.41509433962264

# "services" performs way better than "connect" - a combined 0% drop-off/homepage-return rate compared to 26.4%.

# FINAL CONCLUSION: "SERVICES" shows the best CTR of all options (although when compared to "CONNECT" not statistically significant) and has a ZERO PERCENT drop-off/homepage-return-rate. Therefore it is strongly advised to change the library's homepage and roll out the design with the "SERVICES"-button.